mono/utils/mono-error-internals.h

```diff
@@ -77,6 +80,15 @@ void
 mono_error_set_generic_error (MonoError *error, const char * name_space, const char *name, const char *msg_format, ...);
 
 void
+mono_error_set_exception_instance (MonoError *error, MonoException *exc);
+
+gboolean
+mono_error_is_exception_instance (MonoError *error);
+
+MonoException*
+mono_error_extract_exception_instance (MonoError *error);
```

Comment:
Using either of the last 2 functions would be be a code smell.

We already have mono_error_prepare_exception and mono_error_convert_to_exception for that.